src/main/java/com/michelin/kafkactl/command/Delete.java

```diff
@@ -109,32 +113,54 @@ private List<Resource> parseResources(String namespace) {
         }
 
         // Generate a single resource with minimum details from input
-        return List.of(Resource.builder()
+        var builder = Resource.builder()
             .metadata(Metadata.builder()
                 .name(config.nameConfig.name)
                 .namespace(namespace)
                 .build())
-            .kind(optionalApiResource.get().getKind())
-            .build());
+            .kind(optionalApiResource.get().getKind());
+
+        if (config.nameConfig.version.isPresent()) {
+            builder = builder.spec(Map.of(VERSION, config.nameConfig.version.get()));
+        }
+        return List.of(builder.build());
     }
 
-    static class EitherOf {
+    /**
+     * By-name of by-file deletion config.
+     */
+    public static class EitherOf {
+        /**
+         * Configuration for deletion by name.
+         */
         @ArgGroup(exclusive = false)
         public ByName nameConfig;
 
+        /**
+         * Configuration for deletion by file.
+         */
         @ArgGroup(exclusive = false)
         public ByFile fileConfig;
     }
 
-    static class ByName {
+    /**
+     * By-name deletion config.
+     */
+    public static class ByName {
         @Parameters(index = "0", description = "Resource type.", arity = "1")
         public String resourceType;
 
         @Parameters(index = "1", description = "Resource name.", arity = "1")
         public String name;
+
+        @Option(names = {"-V", "--version"}, description = "Version of the resource to delete.", arity = "0..1")
```

Comment:
Just wondering if `--version` should be included in `ByName`, and what should happen if `kafkactl delete -f schema.yml --version 3` is executed.

Also I just realized that `--version` and `-V` are already reserved for the current version of Kafkactl.. 😄

Reply:
As discussed, `--version` suits the ByName configuration. 
We can include the handling of version in the ByFile way with the `spec.version` field though: 
- if the `spec.version` is specified in the file, we delete the specifided version.
- if not, we delete all the versions of the subject

Regarding the `--version` and `-V`, it is indeed used with piccocli to display the version of the client, it seems like defining that same flag just overloaded the flag for the very specific command `kafkactl delete <resource> <resourceName>`
I need to check if it is possible to bind the displaying of version to the command `kafkactl -V` only.